lib/pbench/server/templates.py

```diff
@@ -256,13 +146,276 @@ def __init__(self, basepath, idx_prefix, logger, known_tool_handlers=None, _dbg=
             "template_name": "{prefix}.v{version}.{idxname}",
             "template_pat": "{prefix}.v{version}.{idxname}.*",
             "template": "{prefix}.v{version}.{idxname}.{year}-{month}-{day}",
+            "owned": True,
             "desc": "Daily tool data for all tools land in indices"
             " named by tool; e.g. prefix.v0.tool-data-iostat.YYYY-MM-DD",
         },
     }
 
+    _fpat = re.compile(r"tool-data-frag-(?P<toolname>.+)\.json")
+
+    def __init__(
+        self,
+        prefix: str = None,
```

Comment:
I think I saw a spot where the code was omitting the `prefix` argument, so a default value is needed, but shouldn't it be `""` rather than `None`?

Reply:
A value of `None` when formatted as a string will yield `"None"` which is probably not what you want....

Reply:
OK, with the second commit, I no longer see a call where `prefix` is unspecified; so I think you probably should make this be a required parameter and object if it comes in with a "False" value.

Reply:
I think I was trying to make clear that the parameters could be referenced by keyword. In retrospect, that doesn't really help. I'm not going to add null checks everywhere, because that seems wasted effort, but I will remove the `= None` where it's not appropriate.